rcode-reader/src/main/java/eu/rssw/pct/RCodeInfo.java

```diff
@@ -0,0 +1,723 @@
+    private Collection<PropertyElement> properties = new ArrayList<>();
+    private Collection<EventElement> events = new ArrayList<>();
+    private Collection<VariableElement> variables = new ArrayList<>();
+    private Collection<TableElement> tables = new ArrayList<>();
+    private Collection<BufferElement> buffers = new ArrayList<>();
+
+    public Function getMainProcedure() {
+      return mainProcedure;
+    }
+
+    public Collection<Function> getFuncs() {
+      return funcs;
+    }
+
+    public Collection<MethodElement> getMethods() {
+      return methods;
+    }
+
+    public Collection<PropertyElement> getProperties() {
+      return properties;
+    }
+
+    public Collection<EventElement> getEvents() {
+      return events;
+    }
+
+    public Collection<VariableElement> getVariables() {
+      return variables;
+    }
+
+    public Collection<TableElement> getTables() {
+      return tables;
+    }
+
+    public Collection<BufferElement> getBuffers() {
+      return buffers;
+    }
+  }
+  
+  public static class Function {
+    private EnumSet<SigAccessType> accessTypes;
+    private String name;
+    private List<Parameter> parameters = new ArrayList<>();
+    private DataType returnType;
+    private String classReturnType;
+    private int returnExtent;
+    private FunctionType type;
+
+    @Override
+    public String toString() {
+      StringBuilder sb = new StringBuilder();
+      for (SigAccessType type : accessTypes) {
+        sb.append(type).append(' ');
+      }
+      sb.append(type).append(' ').append(returnType);
+      if (returnType == DataType.CLASS) {
+        sb.append('{').append(classReturnType).append('}');
+      }
+      if (returnExtent > 0) {
+        sb.append(" EXTENT ").append(returnExtent);
+      }
+      sb.append(' ').append(name).append(" (");
+
+      StringBuilder sb2 = new StringBuilder();
+      for (Parameter p : parameters) {
+        if (sb2.length() > 0) {
+          sb2.append(", ");
+        }
+        sb2.append(p.toString());
+      }
+      sb.append(sb2.toString()).append(')');
+
+      return sb.toString();
+    }
+  }
+
+  public enum FunctionType {
+    MAIN,
+    CONSTRUCTOR,
+    METHOD,
+    FUNCTION,
+    PROCEDURE,
+    EXTERNAL_PROCEDURE,
+    DLL_PROCEDURE,
+    DESTRUCTOR;
+
+    public static FunctionType getTypeFromString(String str) {
+      switch (str.toUpperCase()) {
+        case "CONST": return CONSTRUCTOR;
+        case "MAIN": return MAIN;
+        case "METH": return METHOD;
+        case "FUNC": return FUNCTION;
+        case "PROC": return PROCEDURE;
+        case "EXT": return EXTERNAL_PROCEDURE;
+        case "DLL": return DLL_PROCEDURE;
+        case "DEST": return DESTRUCTOR;
+      }
+
+      return null;
+    }
+  }
+
+  public enum SigAccessType {
+    PUBLIC,
+    PRIVATE,
+    PROTECTED,
+    STATIC,
+    ABSTRACT,
+    FINAL,
+    OVERRIDE;
+
+    public static EnumSet<SigAccessType> getTypeFromString(String str) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "Set" rather than the implementation "EnumSet". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1319)